src/librustc_typeck/check/wfcheck.rs

```diff
@@ -275,6 +278,95 @@ pub fn check_impl_item(tcx: TyCtxt<'_>, def_id: LocalDefId) {
     check_associated_item(tcx, impl_item.hir_id, impl_item.span, method_sig);
 }
 
+fn check_param_wf(tcx: TyCtxt<'_>, param: &hir::GenericParam<'_>) {
+    match param.kind {
+        // We currently only check wf of const params here.
+        hir::GenericParamKind::Lifetime { .. } | hir::GenericParamKind::Type { .. } => (),
+
+        // Const parameters are well formed if their
+        // type is structural match.
+        hir::GenericParamKind::Const { ty: hir_ty } => {
+            let ty = tcx.type_of(tcx.hir().local_def_id(param.hir_id));
+
+            let err_ty_str;
+            let err = if tcx.features().min_const_generics {
+                match ty.kind {
+                    ty::Bool | ty::Char | ty::Int(_) | ty::Uint(_) | ty::Error(_) => None,
+                    ty::FnPtr(_) => Some("function pointers"),
+                    ty::RawPtr(_) => Some("raw pointers"),
+                    _ => {
+                        err_ty_str = format!("`{}`", ty);
```

Comment:
```suggestion
                        err_ty_str = format!("`{}`s", ty);
```
Maybe this reads a little better? (Or, alternatively, changing the text `"using {} as const generic parameters is forbidden"` to `"{} is forbidden as the type of a const generic parameter"` for the non-`FnPtr`/`RawPtr` case.)

Reply:
Implemented the second suggestion, also moved "note: more complex types are supported with `#[feature(const_generics)]`" to non trigger for pointers.